src/builders_v2/linux_vm/nvidia.rs

```diff
@@ -292,29 +279,59 @@ impl Step<LinuxVMBuildContext> for BuildDrivers {
         info!("building NVIDIA drivers");
         let kernel = ctx.get::<Kernel>("kernel").expect("kernel");
 
-        // TODO: should we return error in this case or not?
-        if kernel.is_precompiled() {
-            warn!("Installing NVIDIA drivers for precompiled kernel is not supported yet!");
-            warn!("Skipping installation.");
-            return Ok(());
+        match kernel {
+            Kernel::Precompiled { .. } => {
+                // TODO: should we return error in this case or not?
+
+                warn!("Installing NVIDIA drivers for precompiled kernel is not supported yet!");
+                warn!("Skipping installation.");
```

Comment:
Yes, I think the VM build must fail if the given orders cannot be fulfilled. If user asked for GPU drivers w/ precompiled kernel and we cannot produce such combination, then the program should fail. This saves time from the end user as [s]he doesn't need to debug why the GPU workload doesn't work :slightly_smiling_face:

Reply:
So I did it to match the behavior of v1 builder, which was just warning the user. The only reason why I'm hesitating here is because we could (and probably should) add an option like `--kernel-sources` which will provide NVIDIA driver builder with all it needs. However I haven't implemented it yet. Because of that let's switch this to error, yeah. It's gonna simplify things for user.